api/src/services/import-services/critter/import-critters-service.ts

```diff
@@ -0,0 +1,306 @@
+   */
+  async _getCSVConfig(): Promise<CSVConfig<CritterCSVStaticHeader>> {
+    const [tsnHeaderConfig, aliasHeaderConfig, sexHeaderConfig, dynamicHeadersConfig] = await Promise.all([
+      this._getTsnHeaderConfig(),
+      this._getAliasHeaderConfig(),
+      this._getSexHeaderConfig().catch(() => undefined), // If this throws due to invalid TSNs, we can ignore this header till TSNs are fixed
+      this._getCollectionUnitDynamicHeaderConfig().catch(() => undefined) // Same for the dynamic columns
+    ]);
+
+    return merge(this._config, {
+      staticHeadersConfig: {
+        ITIS_TSN: tsnHeaderConfig,
+        ALIAS: aliasHeaderConfig,
+        SEX: sexHeaderConfig
+      },
+      dynamicHeadersConfig: dynamicHeadersConfig,
+      ignoreDynamicHeaders: dynamicHeadersConfig ? false : true
+    });
+  }
+
+  /**
+   * Get the Critterbase and SIMS import payloads.
+   *
+   * @param {CSVRowValidated[]} rows - The validated CSV rows
+   * @returns {*} { simsPayload: string[]; critterbasePayload: IBulkCreate } The import payloads
+   */
+  _getImportPayloads(rows: CSVRowValidated<CritterCSVStaticHeader>[]): {
+    simsPayload: string[];
+    critterbasePayload: IBulkCreate;
+  } {
+    const simsPayload: string[] = [];
+    const critterbasePayload: IBulkCreate = { critters: [], collections: [] };
+
+    // Convert rows to Critterbase and SIMS payloads
+    for (const row of rows) {
+      const critterId = v4();
+
+      // SIMS payload
+      simsPayload.push(critterId);
+
+      // Critterbase static headers payload
+      critterbasePayload.critters?.push({
+        critter_id: critterId,
+        sex_qualitative_option_id: row.SEX,
+        itis_tsn: row.ITIS_TSN,
+        animal_id: row.ALIAS,
+        wlh_id: row.WLH_ID,
+        critter_comment: row.DESCRIPTION
+      });
+
+      // Critterbase dynamic headers payload
+      this.configUtils.worksheetDynamicHeaders.forEach((header) => {
+        if (row[header]) {
+          critterbasePayload.collections?.push({
+            collection_unit_id: row[header],
+            critter_id: critterId
+          });
+        }
+      });
+    }
+
+    defaultLog.debug({ label: 'critter import payloads', simsPayload, critterbasePayload });
+
+    return { simsPayload, critterbasePayload };
+  }
+
+  /**
+   * Get the TSN header config.
+   *
+   * Validation rules:
+   *  1. TSN must be a number
+   *  2. TSN must be a real ITIS TSN
+   *
+   * @returns {*} {Promise<CSVHeaderConfig>} The TSN header config
+   */
+  async _getTsnHeaderConfig(): Promise<CSVHeaderConfig> {
+    const rowTsns = this.configUtils.getUniqueCellValues('ITIS_TSN');
+    const taxonomy = await this.platformService.getTaxonomyByTsns(rowTsns);
+    const allowedTsns = new Set(taxonomy.map((taxon) => taxon.tsn));
+
+    return {
+      validateCell: getTsnCellValidator(allowedTsns)
+    };
+  }
+
+  /**
+   * Get the CSV Alias header config.
+   *
+   * Validation rules:
+   *  1. Alias must be a string
```

Comment:
Alias can be a number if the animal is named "2", for example

Reply:
Updated